src/ui/src/App.tsx

```diff
@@ -18,11 +19,40 @@ import ExecutionsIndex from 'pages/app/executions/Index';
 import MonitorsIndex from 'pages/app/monitors/Index';
 import MonitorsDetail from 'pages/app/monitors/Detail';
 import MetricsDetail from 'pages/app/metrics/Detail';
+import OnboardingGettingStarted from 'pages/app/onboarding/GettingStarted';
+
+function RequireSetup({ children }: { children: JSX.Element }) {
+  const [isSetup, setIsSetup] = useState(true);
```

Comment:
We may need to set this to false initially. Were you seeing a flash when starting up the application? Asking because this may be happening. The best way to solve this would be to add a loading screen, but that's more lift.

Reply:
I figured that you would actually see a flash /every time after/ you set up the application if this is false initially - thought about this as well